src/main/java/bio/terra/common/exception/DataRepoException.java

```diff
@@ -50,7 +51,9 @@ public String toString() {
         if (errorDetails == null) {
             return super.toString();
         }
-        String details = errorDetails.stream().collect(Collectors.joining("; "));
-        return super.toString() + " Details: " + details;
+        try (Stream<String> stream = errorDetails.stream()) {
```

Comment:
Can you comment why the `try` is added here?

Reply:
Yes, so this is a pattern I've used when working with `Stream`.  When you create a stream from a collection, you have to remember that it's a `Closable` object which means it needs to be closed.  Java 8 introduced `Autoclosable` along with try with resource which handles the closing.  What happens is that you put the `Autoclosable` instantiation in the `...` part of ```try(...){//do the thing}``` and java takes care of closing it. It's much more concise than having to do it in a try...catch...finally block.

Now, the thing that surprised me about this when I found out was that the terminal operation on the stream (e.g. the part that collects) doesn't close the stream, even though you can't do anything with it.  This is - and warning, this is just my opinion - a huge design flaw in the stream system in Java and why I personally don't recommend using it.

In this particular case, not having closed the stream wouldn't be a huge deal since it's a stream built on a List and so will be dereferenced as soon as the list is dereferenced.  This becomes really bad when you use a stream based on, say, a File read (e.g. ```java.nio.file.Files#lines()```) since this will keep a reference to your file and that becomes a resource leak.

All this to say, if we're using streams, I'd use this pattern everywhere to not have to rationalize what the base implementation of the Collection is. I'd just use it and be safe.  Or use a library like Eclipse collections which provides (I think) a much nicer implementation and faster implementation of Java's collections.

Reply:
Fair point about the string join.  For some reason, I was thinking that it limited us to a `char` as the separator, but clearly not.  Will update

Reply:
Though my broader point still stands that we need to be really careful about Streams